File: backend/src/main/kotlin/com/tkppp/sportresult/kbo/domain/KboMatch.kt
package com.tkppp.sportresult.kbo.domain

import com.tkppp.sportresult.kbo.util.KboTeam
import com.tkppp.sportresult.kbo.util.MatchProgress
import com.tkppp.sportresult.kbo.util.MatchStatus
import org.hibernate.annotations.ColumnDefault
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime
import javax.persistence.*

@Entity
class KboMatch(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    val matchDate: LocalDate,

    @Column(nullable = true)
    val startTime: LocalTime? = null,

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    var matchStatus: MatchStatus,

    @Column(nullable = true)
    @Enumerated(EnumType.STRING)
    var matchProgress: MatchProgress? = null,

    @Column(nullable = true)
    @Enumerated(EnumType.STRING)
    val home: KboTeam?,

    @Column(nullable = true)
    @Enumerated(EnumType.STRING)
    val away: KboTeam?,

    @Column(nullable = false)
    var homeScore: Int = 0,

    @Column(nullable = false)
    var awayScore: Int = 0,
) {
}

File: node-backend/src/domain/kbo/kboMatch.ts
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
import { MatchStatus, MatchProgress, Team } from './kbo.utils';

@Entity()
export class KboMatch {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'date', nullable: false })
  matchDate: Date;

  @Column({ type: 'time', nullable: false })
  startTime: Date;

  @Column({
    type: 'enum',
    enum: MatchStatus,
    default: MatchStatus.BEFORE_MATCH,
    nullable: false,
  })
  matchStatus: MatchStatus;

  @Column({
    type: 'enum',
    enum: MatchProgress,
    default: null,
    nullable: true,
  })
  matchProgress: MatchProgress;

  @Column({
    type: 'enum',
    enum: Team,
    nullable: false,
  })
  home: Team;

  @Column({
    type: 'enum',
    enum: Team,
    nullable: false,
  })
  away: Team;

  @Column({
    default: 0,
    nullable: false,
  })
  homeScore: number;

  @Column({
    default: 0,
    nullable: false,
  })
  awayScore: number;
}
